admin/services/group.service.js

```diff
@@ -93,12 +93,11 @@ groupService.create = async (groupName, groupPupils, schoolId) => {
  * @returns {Promise<*>}
  */
 groupService.filterGroupsByPupil = async (schoolId, pupils) => {
-  let pupilIds = ''
-  if (pupils.length > 0) {
-    pupils.map(p => { if (p.group_id) pupilIds += "'" + p.group_id + "', " })
-    pupilIds = pupilIds.length > 0 ? pupilIds.substr(0, pupilIds.length - 2) : ''
+  if (pupils.length < 1) {
+    return false
```

Comment:
is returning `false` significant in any way?  if not, just return, or just dont bother calling this method in the first place if there are no pupils to process...